diff --git a/arch/arm64/configs/cepheus_defconfig b/arch/arm64/configs/cepheus_defconfig
index 4cd02984d..c18e5e1cd 100644
--- a/arch/arm64/configs/cepheus_defconfig
+++ b/arch/arm64/configs/cepheus_defconfig
@@ -204,6 +204,9 @@ CONFIG_SYSFS_SYSCALL=y
 CONFIG_POSIX_TIMERS=y
 CONFIG_KALLSYMS=y
 CONFIG_KALLSYMS_ALL=y
+# ReSukiSU
+CONFIG_KSU=y
+CONFIG_KSU_MANUAL_HOOK=y
 # CONFIG_KALLSYMS_ABSOLUTE_PERCPU is not set
 CONFIG_KALLSYMS_BASE_RELATIVE=y
 CONFIG_PRINTK=y
diff --git a/fs/stat.c b/fs/stat.c
index 220992b55..29cd1df27 100644
--- a/fs/stat.c
+++ b/fs/stat.c
@@ -169,7 +169,7 @@ EXPORT_SYMBOL(vfs_statx_fd);
 
 // KernelSU hook
 #ifdef CONFIG_KSU
-extern int ksu_handle_stat(int *dfd, const char __user **filename_user);
+extern int ksu_handle_stat(int *dfd, const char __user **filename_user, int *flags);
 #endif
 
 /**
@@ -195,7 +195,7 @@ int vfs_statx(int dfd, const char __user *filename, int flags,
 	unsigned int lookup_flags = LOOKUP_FOLLOW | LOOKUP_AUTOMOUNT;
 
 #ifdef CONFIG_KSU
-        ksu_handle_stat(&dfd, &filename); // call KSU hook first
+        ksu_handle_stat(&dfd, &filename, &flags); // call KSU hook first
 #endif
 
 	if ((flags & ~(AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT |
